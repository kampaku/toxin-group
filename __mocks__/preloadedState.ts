const currentRooms = [
  {
    accessibility: { wideCorridor: true, assistant: false },
    additionalFacilities: {
      breakfast: true,
      crib: false,
      feedingChair: false,
      shampoo: false,
      tv: false,
      writingDesk: false,
    },
    freeDays: {
      from: '2022-10-11T06:17:02.467Z',
      to: '2022-10-18T00:20:15.895Z',
    },
    guests: { kids: 2, baby: 0, adults: 4 },
    id: 10,
    imgSrc: [
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
    ],
    number: 111,
    price: 11423,
    rating: 5,
    reviewsAmount: 99,
    roomAmenities: { bedrooms: 2, beds: 4, bathrooms: 0 },
    rules: { allowSmoke: false, allowGuests: true, allowPets: true },
    type: 'Люкс',
  },
  {
    accessibility: { wideCorridor: true, assistant: false },
    additionalFacilities: {
      breakfast: true,
      crib: false,
      feedingChair: false,
      shampoo: false,
      tv: false,
      writingDesk: false,
    },
    freeDays: {
      from: '2022-10-11T06:17:02.467Z',
      to: '2022-10-18T00:20:15.895Z',
    },
    guests: { kids: 2, baby: 0, adults: 4 },
    id: 20,
    imgSrc: [
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
      'https://firebasestorage.googleapis.com/v0/b/toxin-…=media&token=43bbc6bb-a9a8-4731-ba0e-89784af2195e',
    ],
    number: 222,
    price: 11423,
    rating: 5,
    reviewsAmount: 99,
    roomAmenities: { bedrooms: 2, beds: 4, bathrooms: 0 },
    rules: { allowSmoke: true, allowGuests: true, allowPets: false },
    type: 'Люкс',
  },
];

const filters = {
  accessibility: {
    wideCorridor: false,
    assistant: false,
  },
  additionalFacilities: {
    breakfast: false,
    crib: false,
    feedingChair: false,
    shampoo: false,
    tv: false,
    writingDesk: false,
  },
  freeDays: {
    from: null,
    to: null,
  },
  guests: { kids: 0, adults: 0, baby: 0 },
  price: { from: 5000, to: 10000 },
  roomAmenities: { bedrooms: 0, beds: 0, bathrooms: 0 },
  rules: { allowGuests: false, allowSmoke: false, allowPets: false },
};

const user = {
  id: '',
  name: '',
  surname: '',
  email: '',
  birthday: '',
  gender: '',
  getSpecOffers: false,
  likes: [],
};
  
const booking = {
  data: [
    {
      date: { from: '21.08.2022', to: '31.08.2022' },
      guests: { adults: 2, baby: 2, kids: 2 },
      totalCost: 2000,
      imgSrc: [''],
      number: 123,
      price: 20000,
      rating: 3,
      reviewsAmount: 10,
      type: '',
    },
  ],
  status: 'fulfilled',
};

export const preloadedState = {
  rooms: { current: currentRooms },
  searchFilters: filters,
  user,
  booking
};
