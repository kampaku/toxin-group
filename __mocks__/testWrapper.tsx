import React, { PropsWithChildren } from 'react';
import { configureStore, PreloadedState } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { NextRouter } from 'next/router';
import { RouterContext } from 'next/dist/shared/lib/router-context';
import { render, RenderOptions } from '@testing-library/react';

import { createMockRouter } from './createMockRouter';

import { combinedReducer, RootState } from '../src/redux/store';

const setupStore = (preloadedState?: PreloadedState<RootState>) => {
  return configureStore({
    reducer: combinedReducer,
    preloadedState,
  });
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: ReturnType<typeof setupStore>;
}

const renderWithProviders = (
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  router: Partial<NextRouter> = {}
) => {
  const Wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <Provider store={store}>
        <RouterContext.Provider value={createMockRouter(router)}>
          {children}
        </RouterContext.Provider>
      </Provider>
    );
  };
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

export { renderWithProviders };
