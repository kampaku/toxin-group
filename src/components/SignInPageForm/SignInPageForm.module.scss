@import '../../scss/utils/vars';
@import '../../scss/utils/font-template';

.signInPageForm {
  max-width: 380px;
  width: 100%;
  max-height: 358px;
  height: 100%;
  margin: 0px auto;
  background: $colorBackground;
  border: 1px solid rgba(0, 0, 0, 0.12);
  box-shadow: 0px 0px 25px rgba(0, 0, 0, 0.2);
  border-radius: 4px;

  &__inner {
    width: 100%;
    height: 100%;
    padding: 30px;
  }

  &__title {
    @extend %sub-section-title;
    margin: 0px;
    padding: 10px 0px;
  }

  &__input {
    margin: 10px 0px;
  }

  &__button {
    max-width: 320px;
    width: 100%;
    height: 44px;
    margin: 20px 0px;
  }

  &__direction {
    margin-top: 30px;
    display: flex;
    flex-direction: row;
    gap: 10px;
    justify-content: space-between;
    align-items: center;
  }

  &__text {
    @extend %body-text;
    margin: 0px;
  }

  &__link {
    width: 99px;
    height: 44px;
  }
}
