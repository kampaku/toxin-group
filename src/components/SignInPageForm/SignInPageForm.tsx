import { FC, FormEvent, useState } from 'react';
import { useTranslation } from 'next-i18next';

import { Button, Input } from 'components';

import styles from './SignInPageForm.module.scss';

type Props = {
  handleOnSubmit: (email: string, password: string) => void;
};

const SignInPageForm: FC<Props> = ({ handleOnSubmit }) => {
  const [email, setEmail] = useState('');
  const { t } = useTranslation('sign-in-form');
  const handleInputEmailOnChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setEmail(e.target.value);
  };

  const [password, setPassword] = useState('');
  const handleInputPasswordOnChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    setPassword(e.target.value);
  };

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault();
    handleOnSubmit(email, password);
  };

  return (
    <form className={styles.signInPageForm} onSubmit={handleSubmit}>
      <div className={styles.signInPageForm__inner}>
        <h2 className={styles.signInPageForm__title}>{t('title')}</h2>
        <div className={styles.signInPageForm__input}>
          <Input type="email" onChange={handleInputEmailOnChange} />
        </div>
        <div className={styles.signInPageForm__input}>
          <Input
            placeholder={t('password')}
            type="password"
            onChange={handleInputPasswordOnChange}
          />
        </div>
        <div className={styles.signInPageForm__button}>
          <Button type="submit">{t('buttonSignIn')}</Button>
        </div>
        <div className={styles.signInPageForm__direction}>
          <p className={styles.signInPageForm__text}>{t('account')}</p>
          <div className={styles.signInPageForm__link}>
            <Button type="signIn" theme="authorization" href="/Registration">
              {t('buttonSignUp')}
            </Button>
          </div>
        </div>
      </div>
    </form>
  );
};

export { SignInPageForm };
