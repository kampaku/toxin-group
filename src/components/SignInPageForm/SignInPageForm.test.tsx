import { fireEvent, screen } from '@testing-library/react';

import { renderWithProviders } from '../../../__mocks__/testWrapper';
import { SignInPageForm } from './SignInPageForm';

describe('Sign-in page', () => {
  let form: HTMLElement;
  const handeOnSubmit = jest.fn();

  beforeEach(() => {
    form = renderWithProviders(
      <SignInPageForm handleOnSubmit={handeOnSubmit} />,
      { preloadedState: {} },
      { asPath: '/' }
    ).container;
  });

  test('Sign in snapshot', () => {
    expect(form).toMatchSnapshot();
  });

  test('Email input exist', () => {
    const emailInput = screen.getByPlaceholderText(/Email/i);
    expect(emailInput).toBeInTheDocument();
  });

  test('Email input work correctly', () => {
    const emailInput = screen.getByPlaceholderText(/Email/i);
    fireEvent.change(emailInput, {
      target: { value: 'test@gmail.com' },
    });
    expect(emailInput).toHaveValue('test@gmail.com');
  });

  test('Password input exist', () => {
    const passwordInput = screen.getByPlaceholderText(/Password/i);
    expect(passwordInput).toBeInTheDocument();
  });

  test('Password input work correctly', () => {
    const passwordInput = screen.getByPlaceholderText(/Password/i);
    fireEvent.change(passwordInput, {
      target: { value: '12345678' },
    });
    expect(passwordInput).toHaveValue('12345678');
  });

  test('Sign in button exist', () => {
    const submitButton = screen.getByRole('button');
    expect(submitButton).toHaveAccessibleName('buttonSignIn');
  });

  test('Sign up button exist', () => {
    const registrationButton = screen.getByRole('link');
    expect(registrationButton).toBeInTheDocument();
  });

  test('Sign in button can be called', () => {
    const submitButton = screen.getByRole('button');
    fireEvent.click(submitButton);
    expect(handeOnSubmit).toBeCalled();
  });
});
