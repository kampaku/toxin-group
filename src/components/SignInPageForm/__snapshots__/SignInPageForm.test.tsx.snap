// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign-in page Sign in snapshot 1`] = `
<div>
  <form
    class="signInPageForm"
  >
    <div
      class="signInPageForm__inner"
    >
      <h2
        class="signInPageForm__title"
      >
        title
      </h2>
      <div
        class="signInPageForm__input"
      >
        <div
          class=""
        >
          <input
            class="input"
            name=""
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        class="signInPageForm__input"
      >
        <div
          class=""
        >
          <input
            class="input"
            name=""
            placeholder="password"
            type="password"
            value=""
          />
        </div>
      </div>
      <div
        class="signInPageForm__button"
      >
        <button
          class="button button_type_submit"
          type="submit"
        >
          <span
            class="button__icon"
          >
            <svg
              fill="currentColor"
              height="24px"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="24px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
          </span>
          <span
            class="button_type_signInText"
          >
            buttonSignIn
          </span>
        </button>
      </div>
      <div
        class="signInPageForm__direction"
      >
        <p
          class="signInPageForm__text"
        >
          account
        </p>
        <div
          class="signInPageForm__link"
        >
          <a
            class="button button_type_signIn button_in_authorization"
            href="/Registration"
          >
            <span
              class="button_type_signInText"
            >
              buttonSignUp
            </span>
          </a>
        </div>
      </div>
    </div>
  </form>
</div>
`;
